include/boost/geometry/io/wkt/write.hpp

```diff
@@ -515,40 +596,60 @@ class wkt_manipulator
 
     // Boost.Geometry, by default, closes polygons explictly, but not rings
     // NOTE: this might change in the future!
-    inline wkt_manipulator(Geometry const& g,
+    inline wkt_manipulator(Geometry const& g, int significant_digits = -1,
                            bool force_closure = ! is_ring)
         : m_geometry(g)
+        , m_significant_digits(significant_digits)
         , m_force_closure(force_closure)
     {}
 
     template <typename Char, typename Traits>
     inline friend std::basic_ostream<Char, Traits>& operator<<(
-            std::basic_ostream<Char, Traits>& os,
+            std::basic_ostream<Char, Traits>& out,
             wkt_manipulator const& m)
     {
-        dispatch::devarianted_wkt<Geometry>::apply(os, m.m_geometry, m.m_force_closure);
-        os.flush();
-        return os;
+        dispatch::devarianted_wkt<Geometry>::apply(out, m.m_geometry, m.m_significant_digits, m.m_force_closure);
+        out.flush();
+        return out;
     }
 
 private:
     Geometry const& m_geometry;
+    int m_significant_digits;
     bool m_force_closure;
 };
-
 /*!
 \brief Main WKT-streaming function
 \tparam Geometry \tparam_geometry
 \param geometry \param_geometry
+\param significant_digits Specifies the no of significant digits to use in the output wkt
 \ingroup wkt
 \qbk{[include reference/io/wkt.qbk]}
 */
 template <typename Geometry>
-inline wkt_manipulator<Geometry> wkt(Geometry const& geometry)
+inline wkt_manipulator<Geometry> wkt(Geometry const& geometry, int significant_digits = -1)
 {
     concepts::check<Geometry const>();
 
-    return wkt_manipulator<Geometry>(geometry);
+    return wkt_manipulator<Geometry>(geometry, significant_digits);
+}
+
+
+/*!
+\brief Main WKT-string formulating function
+\tparam Geometry \tparam_geometry
+\param geometry \param_geometry
+\param significant_digits Specifies the no of significant digits to use in the output wkt
+\ingroup wkt
+\qbk{[include reference/io/to_wkt.qbk]}
+*/
+template <typename Geometry>
+inline std::string to_wkt(Geometry const& geometry, int significant_digits = -1)
```

Comment:
I don't like this design with a magic number. Shouldn't there be 2 separate overloads of `to_wkt`?
@mloskot @vissarion what is your opinion?

Reply:
The magic of the number is to encode boolean indicator and actual value with single parameter, to distinguish `<0` for not set (use default) or `>=0` for set (use given).

I don't mind separate overloads.

Reply:
What is the final call then? Should I implement seperate overloads?

Reply:
I'm ok with leaving it as it is or removing the parameter at all forcing the user to use `bg::wkt` if he/she needs something different than the default.

@barendgehrels do you have an opinion about it?

Reply:
Although, I do not have strong opinion, here is my suggestion in order of preference: different overloads, leaving as is, totally removing the parameter.